service/history/replication/raw_task_converter.go

```diff
@@ -89,29 +89,35 @@ func NewSourceTaskConverter(
 func (c *SourceTaskConverterImpl) Convert(
 	task tasks.Task,
 ) (*replicationspb.ReplicationTask, error) {
-	if namespaceEntry, err := c.namespaceCache.GetNamespaceByID(
+	var shouldProcessTask bool
+	namespaceEntry, err := c.namespaceCache.GetNamespaceByID(
 		namespace.ID(task.GetNamespaceID()),
-	); err == nil {
-		shouldProcessTask := false
+	)
+	if err != nil {
+		// if there is error, then blindly send the task, better safe than sorry
+		shouldProcessTask = true
+	}
+
+	if namespaceEntry != nil {
 	FilterLoop:
 		for _, targetCluster := range namespaceEntry.ClusterNames() {
 			if c.clientClusterName == targetCluster {
 				shouldProcessTask = true
 				break FilterLoop
 			}
 		}
-		if !shouldProcessTask {
-			return nil, nil
-		}
 	}
-	// if there is error, then blindly send the task, better safe than sorry
+
+	if !shouldProcessTask {
+		return nil, nil
+	}
 
 	clientShardID := common.WorkflowIDToHistoryShard(task.GetNamespaceID(), task.GetWorkflowID(), c.clientClusterShardCount)
 	if clientShardID != c.clientShardKey.ShardID {
 		return nil, nil
 	}
 
-	ctx, cancel := context.WithTimeout(context.Background(), replicationTimeout)
+	ctx, cancel := newTaskContext(namespaceEntry.Name().String())
```

Comment:
The timeout is a bit different, is that a concern?

Reply:
No. This is just load mutable state. Should be quick.